### PR TITLE
Update to uniffi 0.24

### PR DESCRIPTION
**Pull request by arg0d**

### CODE REVIEW

**Comment by ernestask** [comment]
This would benefit from squashing some of the commits.

**Comment by arg0d** [comment]
> This would benefit from squashing some of the commits.

Dõne

**Review by ernestask — APPROVED** [review]

**Review by Savolro — APPROVED** [review]

File: dotnet-tests/UniffiCS.binding_tests/TestCoverall.cs [hunk 1]
```diff
@@ -142,7 +144,7 @@ public class TestCoverall {
         // Make sure that there is no blocking during concurrent FFI calls.
 
         using (var counter = new ThreadsafeCounter()) {
-            const int WAIT_MILLIS = 10;
+            const int WAIT_MILLIS = 20;
```
[the remark this answers is not in the document]
Comment (arg0d) [reply dotnet-tests/UniffiCS.binding_tests/TestCoverall.cs:147]: :shrug: the test is failing. I think anything below 15 or so is flaky. The wake up accuracy of Thread.Sleep is not perfect, and according to ChatGPT the inaccuracy on Windows is up to 15ms. So these kind of short sleeps are not guaranteed to actually wake up after 1ms, and testing with just 10ms seems to be hitting this innacuracy. Though this is happening on Linux, and on Linux the inaccuracy should be less, so I'm not exactly sure why its happening. Maybe C# `Thread.Sleep` adds some kind of additional inaccuracy.

File: bindgen/templates/CallbackInterfaceRuntime.cs [hunk 2]
```diff
@@ -2,6 +2,12 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */#}
 
+static class UniffiCallbackConstants {
```
[the remark this answers is not in the document]
Comment (arg0d) [reply bindgen/templates/CallbackInterfaceRuntime.cs]: fixed

**Pull request merged**